tests/MongoDB.Driver.Core.Tests/Core/Operations/FindOneAndDeleteOperationTests.cs

```diff
@@ -135,20 +135,27 @@ public void Constructor_should_initialize_object()
         [SkippableTheory]
         [ParameterAttributeData]
         public void Execute_with_hint_should_throw_when_hint_is_not_supported(
+            [Values(0, 1)] int w,
             [Values(false, true)] bool async)
         {
+            var writeConcern = new WriteConcern(w);
             var serverVersion = CoreTestConfiguration.ServerVersion;
             var subject = new FindOneAndDeleteOperation<BsonDocument>(_collectionNamespace, _filter, BsonDocumentSerializer.Instance, _messageEncoderSettings)
             {
-                Hint = new BsonDocument("_id", 1)
+                Hint = new BsonDocument("_id", 1),
+                WriteConcern = writeConcern
             };
 
-            var exception = Record.Exception(() => ExecuteOperation(subject, async));
+            var exception = Record.Exception(() => ExecuteOperation(subject, async, useImplicitSession: true));
```

Comment:
Can you explain why you had to change `useImplicitSession` to `true`?

Reply:
Yes, because otherwise we will have an exception here: [CommandUsingCommandMessageWireProtocol](https://github.com/mongodb/mongo-csharp-driver/blob/44c4d03ebf5fd2bc1cd62431f3cf75c80fbd1b89/src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs#L309)

> `throw new InvalidOperationException("Explicit session must not be used with unacknowledged writes.");`

And it fails on Evergreen: [link](https://evergreen.mongodb.com/task_log_raw/dot_net_driver_unsecure_tests__version~4.4_os~windows_64_topology~replicaset_auth~noauth_ssl~nossl_test_patch_caa2b604298223bdf521f2cf7fdfcd6cd9ed68a4_5e90b3dcc9ec447f60926019_20_04_10_17_58_53/0?type=T#L1236)

> `"Explicit session must not be used with unacknowledged writes."`

Reply:
Thanks. Just wanted to know what the reason was.